app/views/assignments/_form.html.erb

```diff
@@ -18,40 +19,73 @@
   </div>
   <div id="map_file_details" style="display:none;">
-  <%= f.label :mapfile, "Mapping file (csv) [?]", title: "The zip file should contain one folder for each student's code files; folders are named according to student ID; Name your folder as \"skeleton\" for the codes you wish to exclude from the system" %>
+  <%= f.label :mapfile, "Mapping file (csv) \u24D8", title: "If you have a map file that maps the filenames to other attributes such as student's names, github handle etc, upload the file here. Ensure that the file is a csv file and follows the conventions according to the user guide." %>
   <%= f.file_field :mapfile, accept: '.csv' %>
-
+  </div>
+  <div id="file_to_ignore_details" style="display:none;">
+  <%= f.label :filesIgnore, "Files to be ignored \u24D8" , title: "If you want to ignore certain files from the code zip file that you uploaded, add those filenmes under here. \n\nIf you want to ignore filenames that exactly match a certain filename, enter the filename as it is. For instance, entering \"question1\", will mean that all the filenames that are named as \"question1\" (irrespective of their file extensions) will all be ignored. \n\nIf you want to ignore filenames that partially contain a substring, add \"/ \" in front of the filename. For instance, if you enter, \"/question1\", filenames such as \"alice_question1.py\", \"bob_question1Test.c\", \"question1_charlie.java\" will all get ignored. \n\nIf you need to add muliple regex, leave a space in between them. For instance, if you enter \"question1 /question2 \", filesnames that either exactly match \"question1 \" or contain the substring \"question2 \" will all get ignored"  %>
+  <%= f.text_field :filesIgnore %>
+  </div>
+  <div id="file_to_process_details" style="display:none;">
+  <%= f.label :filesProcess, "Files to be only processed \u24D8" , title: "If you want to process and compare only certain files from the code zip file that you uploaded, add those filenmes under here. \n\nIf you want to process filenames that exactly match a certain filename, enter the filename as it is. For instance, entering \"question1\", will mean that all the filenames that are named as \"question1\" (irrespective of their file extensions) will all be processed and the rest will be ignored. \n\nIf you want to only process filenames that partially contain a substring, add \"/ \" in front of the filename. For instance, if you enter, \"/question1\", filenames such as \"alice_question1.py\", \"bob_question1Test.c\", \"question1_charlie.java\" will all get processed and the rest that does not contain the substring will all get ignored. \n\nIf you need to add muliple regex, leave a space in between them. For instance, if you enter \"question1 /question2 \", filesnames that either exactly match \"question1 \" or contain the substring \"question2 \" will all get processed."  %>
+  <%= f.text_field :filesProcess %>
+  </div>
+  <div id="ref_dir_details" style="display:none;">
```

Comment:
Will need to point to documentation about how this works.

Reply:
I will create an issue to update user guide so that I can add documentation for all newly added changes.